packages/client/tests/functional/issues/6578/tests.ts

```diff
@@ -60,8 +60,10 @@ testMatrix.setupTestSuite(
           })
         }
 
-        // This test is asserting that JSON.parse does not throw because quotes are used
-        const params = JSON.parse(paramsString)
+        // This test is asserting that JSON.parse does not throw because quotes are used.
+        // However, we may need to retry if parsing fails because logs haven't arrived yet
+        // and `paramsString` is empty.
+        const params = await waitFor(() => JSON.parse(paramsString))
```

Comment:
On the second thought it might've been simpler and perhaps more explicit to just do

```ts
await waitFor(() => paramsString !== '')

const params = JSON.parse(paramsString)
```

Happy to change it if that's what reviewers would prefer, I don't mind either way.

Reply:
I like this version you suggest because it's more explicit and if something fails it would fail on a specific line (and not on a does multiple things line)

Reply:
Actually it won't work like this, it will have to be something like this instead:

```ts
await waitFor(() => {
  if (paramsString === '') {
    throw new Error('params not received')
  }
})
```

So more explicit but not simpler. Would you still prefer it?

Reply:
I like the explicit approach.